src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -65,4 +72,42 @@ public static void deleteTempTargetS3Bucket() throws IOException {
     BeforeActions.scenario.write("Deleted S3 Target Bucket " + s3TargetBucketName);
     s3TargetBucketName = StringUtils.EMPTY;
   }
+
+  public static void verifyDataIsTransferredToTargetS3Bucket() {
+    String s3Bucket = TestSetupHooks.s3TargetBucketName;
+    boolean flag = false;
+    StringBuilder sb = new StringBuilder();
+    try {
+      for (S3ObjectSummary s3ObjectSummary : S3Client.listObjects(s3Bucket)) {
+        if (s3ObjectSummary.getKey().contains("part")) {
+          flag = true;
+          String content = S3Client.getObject(s3Bucket, s3ObjectSummary.getKey());
+          sb.append(content);
+        }
+      }
+      if (flag) {
+        String path = Paths.get(Objects.requireNonNull(TestSetupHooks.class.getResource
+          ("/" + PluginPropertyUtils.pluginProp("outputCSV"))).getPath()).toString();
+        Assert.assertEquals("Output content should match", readFile(path), sb.toString());
```

Comment:
I don't think these strings are guaranteed to be equal because the part files could be created in any order.

Reply:
for this s3->s3, I think it's fine since it only contains 1 partition.